config/config-api/src/main/java/com/thoughtworks/go/domain/label/PipelineLabel.java

```diff
@@ -56,13 +62,21 @@ private String replaceRevisionsInLabel(Map<CaseInsensitiveString, String> materi
 
     private String lookupMaterialRevision(Matcher matcher,  Map<CaseInsensitiveString, String> materialRevisions) {
         final CaseInsensitiveString material = new CaseInsensitiveString(matcher.group(1));
+        String revision;
```

Comment:
If you use `${env:ABC}` instead of: `${$ABC}` would it help to differentiate between environment variables and material names?

Reply:
It actually wouldn't unless we shoe horn the env vars into the materialRevisions. Even then there'd be problems. For example, there's the problem of potentially cobbling values from the material revisions if an env var has the same name as a material

Reply:
Shifted to using the ${env:ABC} anyway. So I'll resolve this

Reply:
More than differentiate, I believe this is unambiguous in case users decide to use the same name.

Reply:
Well, we could just say: `${MATERIAL}` will still work as normal. Then, either one of `${$ENV}` or `${env:ENV}` is both unambiguous and differentiating enough. They should be equivalent, right? It's just that the `env:` option allows you to do more, such as introduce: `date:` etc.